src/dbnode/storage/bootstrap_test.go

```diff
@@ -90,6 +91,45 @@ func TestDatabaseBootstrapWithBootstrapError(t *testing.T) {
 	require.Equal(t, "an error", result.ErrorsBootstrap[0].Error())
 }
 
+func TestDatabaseInBootstrappingStateWhenBootstrapError(t *testing.T) {
+	ctrl := gomock.NewController(xtest.Reporter{T: t})
+	defer ctrl.Finish()
+
+	opts := DefaultTestOptions()
+	db := NewMockdatabase(ctrl)
+	m := NewMockdatabaseMediator(ctrl)
+	m.EXPECT().DisableFileOpsAndWait()
+	m.EXPECT().EnableFileOps().AnyTimes()
+
+	bsm := newBootstrapManager(db, m, opts).(*bootstrapManager)
+	// Don't sleep.
+	bsm.sleepFn = func(time.Duration) {}
+	bootstrapRetries := 0
```

Comment:
Nit (and sorry if it appears twice, somehow my original comment looks gone):
I'd probably switch it semantically to something like `bootstrapIteration` and then simply increment it in the beginning of the anonymous fn (to avoid defer which requires some additional thinking while comprehending the code).

Reply:
OTOH, I think it would be much simpler just to add those `bsm.state` checks to the existing `TestDatabaseBootstrapWithBootstrapError` test (as it already has the purpose of verifying the behaviour with a single retry after a failure).

Reply:
> OTOH, I think it would be much simpler just to add those `bsm.state` checks to the existing `TestDatabaseBootstrapWithBootstrapError` test (as it already has the purpose of verifying the behaviour with a single retry after a failure).

Sure, will do this.